tests/postgres_tests/test_operations.py

```diff
@@ -0,0 +1,206 @@
+
+        Verify that the operation properly handles the creation of a BRIN index
+        as a spot check, ensuring it does not always use the default (B-tree).
+        """
+        project_state = self.set_up_test_model(self.app_label, index=False)
+        new_state = project_state.clone()
+
+        operation = AddIndexConcurrently('Pony', BrinIndex(fields=['pink'], name='pony_pink_brin_idx'))
+
+        with connection.schema_editor(atomic=False) as editor:
+            operation.database_forwards(self.app_label, editor, project_state, new_state)
+
+        self.assertIndexExists('%s_pony' % self.app_label, ['pink'], index_type='brin')
+
+        with connection.schema_editor(atomic=False) as editor:
+            operation.database_backwards(self.app_label, editor, new_state, project_state)
+
+        self.assertIndexNotExists('%s_pony' % self.app_label, ['pink'])
+
+    def test_add_with_options(self):
+        project_state = self.set_up_test_model(self.app_label, index=False)
+        new_state = project_state.clone()
+
+        operation = AddIndexConcurrently(
+            'Pony',
+            BTreeIndex(
+                fields=['pink'],
+                name='pony_pink_btree_idx',
+                fillfactor=70,
+            )
+        )
+
+        with connection.schema_editor(atomic=False) as editor:
+            operation.database_forwards(self.app_label, editor, project_state, new_state)
+
+        self.assertIndexExists('%s_pony' % self.app_label, ['pink'], index_type='btree')
+
+        with connection.schema_editor(atomic=False) as editor:
+            operation.database_backwards(self.app_label, editor, new_state, project_state)
+
+        self.assertIndexNotExists('%s_pony' % self.app_label, ['pink'])
+
+    def test_concurrent_option(self):
+        project_state = self.set_up_test_model(self.app_label, index=False)
+        new_state = project_state.clone()
+
+        operation = AddIndexConcurrently(
+            'Pony',
+            BTreeIndex(
+                fields=['pink'],
+                name='pony_pink_btree_idx',
+                fillfactor=70,
+                concurrent=True,
+            )
+        )
+
+        with connection.schema_editor(atomic=False) as editor:
+            operation.database_forwards(self.app_label, editor, project_state, new_state)
+
+        self.assertIndexExists('%s_pony' % self.app_label, ['pink'], index_type='btree')
+
+        with connection.schema_editor(atomic=False) as editor:
+            operation.database_backwards(self.app_label, editor, new_state, project_state)
+
+        self.assertIndexNotExists('%s_pony' % self.app_label, ['pink'])
+
+
+@unittest.skipUnless(connection.vendor == 'postgresql', 'RemoveIndexConcurrently is PostgreSQL-specific')
+@modify_settings(INSTALLED_APPS={'append': 'migrations'})
+class RemoveIndexConcurrentlyTests(OperationTestBase):
+    app_label = 'test_concurrently'
+
+    def tearDown(self):
```

Comment:
I know this is a dirty, temporary trick. `pony_test_idx` is not deleted by the `TransactionTestCase` class, and I had a look at `MultiDBOperationTests` which also uses `OperationTestBase` as the base where a few migrations are unapplied inside a test-case.

Reply:
An alternative would be to call `self.addCleanup(editor.remove_index, model, index, concurrently=True)` in tests that require this cleanup.

Reply:
I think the move from `OperationTestBase` was done incorrectly, as `tearDown` and `setUpClass` were deleted. I reinserted them.